### ISSUE TITLE
adminKey usage

### ISSUE THREAD

**Issue by DannyMichaels** [DannyMichaels/LoL-scrims-finder#7]
so right now `scrims` can only be created and edited by admins (which is a good thing), but right now the way it's handled is a bit `incomplete`.

In the front-end: there's a-lot of code that says `process.env.REACT_APP_ADMIN_KEY === currentUser?.adminKey ? dothis : dothat` 

the adminKey is already hosted in the `user` object on the back-end. So do we 100% need that variable in the front-end (for conditionally displaying content)? a-lot of things should also be checked in the back-end, like when making a post request for `scrimCreate`, we need to check for admin key, not just in the front-end.

Maybe we can just add a "isAdmin?" boolean on the user object, and check if it's equal to req.body.adminKey, and just send it to the currentUser context on the front-end.

In short: make adminKey usage more  **elegant**.
